src/main/java/com/google/api/codegen/util/LicenseHeaderUtil.java

```diff
@@ -0,0 +1,110 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.util;
+
+import com.google.api.codegen.ConfigProto;
+import com.google.api.codegen.LanguageSettingsProto;
+import com.google.api.codegen.LicenseHeaderProto;
+import com.google.api.tools.framework.model.Diag;
+import com.google.api.tools.framework.model.DiagCollector;
+import com.google.api.tools.framework.model.SimpleLocation;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.CharStreams;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import javax.annotation.Nullable;
+
+public class LicenseHeaderUtil {
+  public static final String DEFAULT_LICENSE_FILE = "license-header-apache-2.0.txt";
+  public static final String DEFAULT_COPYRIGHT_FILE = "copyright-google.txt";
+
+  private LicenseHeaderProto licenseHeader;
```

Comment:
This is never assigned to outside the create method. Can we add it to the constructor, and make it (and diagCollector) final?

Reply:
I would rather keep `licenseHeader` inside the class because it is never used outside of the class, and `licenseHeader` is populated by a non-trivial assignment. Keeping the `config != null` checks inside this class will keep the calling code cleaner, because it just has to call the constructor, without doing the null checks.

Made `diagCollector` final.